domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -3,11 +3,109 @@ package dagtraversalmanager_test
 import (
 	"github.com/kaspanet/kaspad/domain/consensus"
 	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/model/testapi"
 	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
 	"github.com/kaspanet/kaspad/domain/dagconfig"
 	"testing"
 )
 
+func TestBlockAtDepth(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, tearDown, err := factory.NewTestConsensus(params, false,
+			"TestBlockAtDepth")
+		if err != nil {
+			t.Fatalf("NewTestConsensus: %s", err)
+		}
+		defer tearDown(false)
+
+		// "checkExpectedBlockChainDag" - compare BlockAtDepth to a loop on the SelectedChainIterator on a single chain DAG.
+		checkExpectedBlockChainDag := func(highHash *externalapi.DomainHash, depth uint64) {
+			currentBlock := highHash
+			currentBlockData, err := tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+			if err != nil {
+				t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+			}
+			for i := uint64(0); i <= depth; i++ {
+				if currentBlockData.SelectedParent() == nil {
+					break
+				}
+				currentBlock = currentBlockData.SelectedParent()
+				currentBlockData, err = tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+				if err != nil {
+					t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+				}
+			}
+			expected := currentBlock
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(highHash, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+
+			if !blockHash.Equal(expected) {
+				t.Fatalf("Expected block %s but got %s", expected, blockHash)
+			}
+		}
+
+		// "checkExpectedBlockTwoChildrenSameSelectedParent" checks blockAtDepth on 2 children with the same selectedParent.
```

Comment:
2 children of whom?
I'd simply write "2 blocks with the same selectedParent"

Reply:
Changed, thanks.